tmt/steps/discover/fmf.py

```diff
@@ -453,17 +453,17 @@ def assert_git_url(plan_name: Optional[str] = None) -> None:
                     dirs_exist_ok=True)
 
         # Adjust path and optionally show
-        if path is None or path == '.':
-            path = ''
+        if path is None or path.resolve() == Path.cwd().resolve():
+            path = Path('')
         else:
-            self.info('path', path, 'green')
+            self.info('path', str(path), 'green')
 
         # Prepare the whole tree path and test path prefix
-        tree_path = os.path.join(self.testdir, path.lstrip('/'))
-        if not os.path.isdir(tree_path) and not self.opt('dry'):
+        tree_path = self.testdir / (path.relative_to('/') if path.is_absolute() else path)
+        if not tree_path.is_dir() and not self.opt('dry'):
             raise tmt.utils.DiscoverError(
                 f"Metadata tree path '{path}' not found.")
-        prefix_path = os.path.join('/tests', path.lstrip('/'))
+        prefix_path = Path('/') / 'tests' / path
```

Comment:
I'm afraid we need to make sure the `path` does not start with a slash (as it is done on line 462), otherwise the last path element overwrites everything.

Reply:
Is there a test where I could observe the change? In other words, how did you notice there's an issue?

My current patch would be the following, but I'd like to find out whether it's correct:

```python
prefix_path = Path('/') / 'tests' / (path.relative_to('/') if path.is_absolute() else path)
```

Reply:
Should be fixed and covered by a142cd00.

Reply:
Test passes.